.github/workflows/upmerge.yaml

```diff
@@ -1,3 +1,27 @@
+# This workflow automates the process of upmerging changes from the current release branch to the edge branch.
+# During the course of a release, the release branch is the default branch so that PRs can be immediately
+# brought into the release without waiting for a new release. This workflow merges those changes into 
+# the edge branch so that edge can be used as the basis for the next release branch.
+#
+# This workflow assumes that it is being triggered from the current release branch, but it could be triggered from
```

Comment:
So, this is done right after we do the release, right? Let's say we have **_just_** finished releasing 0.36 and 0.36 becomes the new main branch. We kick this workflow off from the 0.36 branch, right? All the commits from edge gets merged into the new release branch i.e. 0.36 in this instance.

- 0.35 was the main branch
- edge has 6 commits on top of 0.35
- we release 0.36
- now 0.36 branch is the main branch
- this workflow is kicked off from 0.36
- 0.36 gets rebased with the branch created by 0.36

Is this the flow?

Reply:
We do not have a formal document on the release process in this repo, but there is a similar process [documented in the docs repo](https://github.com/radius-project/docs/blob/v0.36/docs/content/contributing/docs/maintainers-docs/index.md). In that process, the steps are like this:

Regular PR process:
- v0.35 is the default (main) branch.
- The edge branch was equivalent to v0.35 when the release was created.
- Changes are PR'd to the v0.35 branch and the edge branch falls behind.

Process for creating a new release:
- Create a new branch from edge.
- Merge changes from v0.35 to the branch just created from edge.
- PR the changes to edge from the branch just created. (When this is complete, edge will be at the same commit as v0.35.)
- Create the new release branch, v0.36, as a branch from edge.
- Update the v0.36 branch to be the default (main) branch.

At that point in time, v0.35, v0.36, and edge will all be at the same commit. v0.36 will be the new default branch. New PRs will go to the v0.36 branch.

The `upmerge.yaml` workflow was created to support the process described above. @willdavsmith please correct anything written above.

```mermaid
%%{init: { 'logLevel': 'debug', 'theme': 'base', 'gitGraph': {'showBranches': true, 'showCommitLabel':true,'mainBranchName': 'edge'}} }%%
gitGraph LR:
    commit id: "Release"
    branch v0.35
    commit id: "PR 1"
    commit id: "PR 2"
    commit id: "PR 3"
    checkout edge
    branch edge-next1
    merge v0.35
    checkout edge
    merge edge-next1
    branch v0.36
    commit id: "PR 4"
    commit id: "PR 5"
    commit id: "PR 6"
    checkout edge
    branch edge-next2
    merge v0.36
    checkout edge
    merge edge-next2
    branch v0.37
    commit id: "PR 7"
    commit id: "PR 8"
    commit id: "PR 9"
```

Reply:
What I was thinking is that once a release branch is created, it shouldn't receive any more commits because that branch is like what we have for that specific release.

Then **edge** starts getting the new commits which will eventually become the new release.

Example:

1. 0.35 (main branch) - **30 commits**.
2. All the new PRs get based off of **edge**.
3. Let's assume just before the 0.36 release, edge has 5 more commits on top of the main branch (0.35 branch) meaning **35 commits**.
4. We do the 0.36 release.
5. 0.36 release (main branch) - **35 commits**.
6. Go back to item 2 all the way to item 4.

Does that make sense? cc/ @willdavsmith @willtsai @rynowak

Reply:
> At that point in time, v0.35, v0.36, and edge will all be at the same commit. v0.36 will be the new default branch. New PRs will go to the v0.36 branch.

I don't see a reason why all 3 branches need to be at the same commit. 0.35 should be behind and edge and 0.36 should be at the same commit at the time of 0.36 release.